### PR TITLE
net-misc/bird: add 2.0.10

### PR DESCRIPTION
**Pull request by alarig**

### CODE REVIEW

File: net-misc/bird/bird-2.0.10.ebuild [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools fcaps
+
+DESCRIPTION="A routing daemon implementing OSPF, RIPv2 & BGP for IPv4 & IPv6"
+HOMEPAGE="https://bird.network.cz"
+SRC_URI="ftp://bird.network.cz/pub/${PN}/${P}.tar.gz"
+LICENSE="GPL-2"
```
Comment (MatthewGentoo) [inline net-misc/bird/bird-2.0.10.ebuild:11]:
```suggestion
LICENSE="GPL-2+"
```

**Review by MatthewGentoo — CHANGES_REQUESTED** [review]
Please could you amend your commit with a [GCO sign-off](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin)?

File: net-misc/bird/bird-2.0.10.ebuild (hunk 1 above)
Comment (alarig) [reply net-misc/bird/bird-2.0.10.ebuild:11]: It’s weird that pkgdev didn’t had it automatically (it worked with repoman), but I force-pushed with it anyway.

File: net-misc/bird/bird-2.0.10.ebuild [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86 ~x64-macos"
+IUSE="+client debug libssh"
+
+RDEPEND="
+	client? ( sys-libs/ncurses:= )
+	client? ( sys-libs/readline:= )
```
Comment (juippis) [inline net-misc/bird/bird-2.0.10.ebuild:19]:
```suggestion
	client? (
		sys-libs/ncurses:=
		sys-libs/readline:=
	)
```

**Review by juippis — APPROVED** [review]
Also a bit weirdly,
```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/sbin/bird
 * /usr/bin/birdc
 * /usr/bin/birdcl
```
If you want to fix these you'll have to take a look into the build system. Most likely they're hardcoding CFLAGS.

**Comment by alarig** [comment]
> Also a bit weirdly,
> 
> ```
>  * QA Notice: Files built without respecting CFLAGS have been detected
>  *  Please include the following list of files in your report:
>  * /usr/sbin/bird
>  * /usr/bin/birdc
>  * /usr/bin/birdcl
> ```
> 
> If you want to fix these you'll have to take a look into the build system. Most likely they're hardcoding CFLAGS.

Bird is mono-threaded and have to keep with potentially a lot of BGP updates (e.g. in a RR setup), so it is very very very … very optimised and it has to be compiled with LTO.

**Comment by juippis** [comment]
But I wonder if you can easily have a fix that uses user's CFLAGS but appends whatever upstream also wants to add with those? How're the CFLAGS in their build system? Usually you have something like `+=` and not hardcoding them.

**Comment by alarig** [comment]
It’s already an append (in the upstream `configure.ac`):
```bash
  if test "$enable_debug" = no; then
    BIRD_CHECK_LTO
  fi

  if test "$bird_cv_c_lto" = yes; then
    CFLAGS="$CFLAGS -flto"
    LDFLAGS="$LDFLAGS -flto=4 -g"
  else
    LDFLAGS="$LDFLAGS -g"
  fi
```
https://gitlab.nic.cz/labs/bird/-/blob/2.0.10/configure.ac#L160

**Pull request closed without merging**